### PR TITLE
Allow configurable prov srv container images

### PR DESCRIPTION
**Pull request by abays**

### CODE REVIEW

File: api/v1beta1/openstackprovisionserver_types.go [hunk 1]
```diff
@@ -26,8 +26,17 @@ type OpenStackProvisionServerSpec struct {
 	Port int `json:"port"`
 	// An optional interface to use instead of the cluster's default provisioning interface (if any)
 	Interface string `json:"interface,omitempty"`
-	// URL RHEL qcow2 image (compressed as gz, or uncompressed)
+	// URL for RHEL qcow2 image (compressed as gz, or uncompressed)
 	BaseImageURL string `json:"baseImageUrl"`
+	// Container image URL for init container that downloads the RHEL qcow2 image (baseImageUrl)
+	// +kubebuilder:default="quay.io/abays/downloader:0.0.3"
```
Comment (stuggi) [inline api/v1beta1/openstackprovisionserver_types.go:32]: should we add those images under openstack-k8s-operators ?

File: api/v1beta1/openstackprovisionserver_types.go [hunk 2]
```diff
@@ -26,8 +26,17 @@ type OpenStackProvisionServerSpec struct {
...
+	DownloaderImageURL string `json:"downloaderImageUrl,omitempty"`
+	// Container image URL for the main container that serves the downloaded RHEL qcow2 image (baseImageUrl)
+	// +kubebuilder:default="quay.io/centos7/httpd-24-centos7:2.4"
```
Comment (stuggi) [inline api/v1beta1/openstackprovisionserver_types.go:35]: same

File: api/v1beta1/openstackprovisionserver_types.go [hunk 3]
```diff
@@ -26,8 +26,17 @@ type OpenStackProvisionServerSpec struct {
...
+	ApacheImageURL string `json:"apacheImageUrl,omitempty"`
+	// Container image URL for the sidecar container that discovers provisioning network IPs
+	// +kubebuilder:default="quay.io/abays/provision-ip-discovery-agent:0.0.1"
```
Comment (stuggi) [inline api/v1beta1/openstackprovisionserver_types.go:38]: same

File: api/v1beta1/openstackprovisionserver_types.go (hunk 1 above)
Comment (abays) [reply api/v1beta1/openstackprovisionserver_types.go:32]: Yes, let's do that.  While we're at it, I am going to rename the `downloader` image to something like `rhel-downloader` to give it more context.

Comment (abays) [reply api/v1beta1/openstackprovisionserver_types.go]: Done

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**